docs/template-development.md

```diff
@@ -0,0 +1,159 @@
+---
+title: "Template development"
+weight: 50
+---
+
+## Minimum template requirements
+
+Let's discuss the minimum components your template requires: `template` directory and `package.json` file.
+
+## 1. `template` directory
+
+The `template` directory stores generated outputs in files. In other words, the generator processes all the files stored in this directory.
+
+```js
+import { File, Text } from "@asyncapi/generator-react-sdk";
+
+export default function({ asyncapi, params, originalAsyncAPI }) {
+return (
+    <File name="asyncapi.md">
+    <Text>This is a markdown file for my application.</Text>
+    <Text>App name is: **{ asyncapi.info().title() }**</Text>
+    </File>
+);
+}
+```
+## 2. `package.json` file
+
+Before the generation process begins, the generator installs the template into its dependencies. A `package.json` is necessary to identify the template name.
+
+The following block shows an example `package.json` file that points to the [React Render Engine](react-render-engine.md) and necessary dependencies:
+
+```json
+{
+  "name": "myTemplate",
+  "generator": {
+    "renderer": "react"
+  },
+  "dependencies": {
+    "@asyncapi/generator-react-sdk": "^0.2.25"
+  }
+}
+```
+
+The above example of a`template/index.js` file shows the generation process result. The user also receives an `asyncapi.md` file with hardcoded and dynamic (title of the application taken from AsyncAPI document) information.
+
+Every template must depend on the [`@asyncapi/generator-react-sdk` package](https://github.com/asyncapi/generator-react-sdk), which contains a template file's basic components.
+
+## Additional configuration options
```

Comment:
this section got long as there are instructions for package.json and template files, maybe under this section add to subsections?

```
### Adding configuration to package.json

### Accessing parameters inside template
```

wdyt?